src/CMakeLists.txt

```diff
@@ -587,15 +585,22 @@ if(${WITH_RADOSGW})
     cls_statelog_client
     cls_rgw_client
     fcgi
-    ${CURL_LIBRARIES}
     ${EXPAT_LIBRARIES}
   )
 endif(${WITH_RADOSGW})
-if(${WITH_RBD})
+if(WITH_RBD)
   set(DENCODER_EXTRALIBS
       ${DENCODER_EXTRALIBS}
       rbd_replay_types)
-endif(${WITH_RBD})
+  if(LINUX)
+    list(APPEND dencoder_srcs
+        $<TARGET_OBJECTS:krbd_objs>)
+  endif(LINUX)
+endif(WITH_RBD)
+if(WITH_KVS)
+  list(APPEND dencoder_srcs
+      $<TARGET_OBJECTS:parse_secret_objs>)
```

Comment:
@tchaikov 
I think it is like that because that is how it got out of the autobuild - Cmake conversion.
But you are correct that only krbd and mount.ceph reference secret.h.
I even checked calls to any of the 3 functions defined in secret.h.

However secret.c is added to the target: parse_secret_files
Which is used in:
```
./src/tools/rbd_nbd/CMakeLists.txt:2:  $<TARGET_OBJECTS:parse_secret_objs>)
./src/tools/rbd/CMakeLists.txt:41:  $<TARGET_OBJECTS:parse_secret_objs>
./src/rbd_replay/CMakeLists.txt:17:  $<TARGET_OBJECTS:parse_secret_objs>
./src/rbd_replay/CMakeLists.txt:32:    $<TARGET_OBJECTS:parse_secret_objs>
./src/CMakeLists.txt:574:  $<TARGET_OBJECTS:parse_secret_objs>
./src/CMakeLists.txt:757:set(parse_secret_files
./src/CMakeLists.txt:759:add_library(parse_secret_objs OBJECT ${parse_secret_files})
./src/CMakeLists.txt:790:    $<TARGET_OBJECTS:parse_secret_objs>
./src/test/librbd/CMakeLists.txt:55:  $<TARGET_OBJECTS:parse_secret_objs>
./src/test/librbd/CMakeLists.txt:88:  $<TARGET_OBJECTS:parse_secret_objs>
./src/test/librbd/CMakeLists.txt:114:  $<TARGET_OBJECTS:parse_secret_objs>
./src/test/librbd/CMakeLists.txt:131:  $<TARGET_OBJECTS:parse_secret_objs>
./src/test/bench/CMakeLists.txt:25:    $<TARGET_OBJECTS:parse_secret_objs>
./src/test/CMakeLists.txt:779:  $<TARGET_OBJECTS:parse_secret_objs>
./src/test/cls_rbd/CMakeLists.txt:5:  $<TARGET_OBJECTS:parse_secret_objs>
./src/test/rbd_mirror/CMakeLists.txt:81:  $<TARGET_OBJECTS:parse_secret_objs>)
./src/librbd/CMakeLists.txt:78:  $<TARGET_OBJECTS:parse_secret_objs>
```

So your suggestion is that most of those references could/should go?
And just about everything should be able to build without parse_secrets_obj.

But that would not be correct to do in this PR.
But I'll see if there is another way will work in a Linux build.